nixos/modules/hardware/network/ath-user-regd.nix

```diff
@@ -0,0 +1,31 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  kernelVersion = config.boot.kernelPackages.kernel.version;
+  linuxKernelMinVersion = "5.8";
+  kernelPatch = pkgs.kernelPatches.ath_regd_optional // {
+    extraConfig = ''
+      ATH_USER_REGD y
+    '';
```

Comment:
Can you use `structuredExtraConfig` instead? Like this: https://github.com/NixOS/nixpkgs/blob/bf0589a3c5f049c2aafafcc0bd5d32119d00ddc2/pkgs/os-specific/linux/kernel/linux-rt-5.4.nix#L28-L36

Reply:
Thanks for your comment! Your example, however, refers to a derivation of an entire kernel rather than an implementation of a mere patch. Therefore, I'm not sure how to adopt this approach. At least, [according to the documentation](https://github.com/NixOS/nixpkgs/blob/bf0589a3c5f049c2aafafcc0bd5d32119d00ddc2/doc/builders/packages/linux.section.md), a `kernelPatch` doesn't have an attribute `structuredExtraConfig`. Could you elaborate please?